File: 0075-sort-colors.cpp
/*
75. Sort Colors

Submitted: December 20, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 11.46 MB (beats 35.93%)
*/

class Solution {
public:
    // counting sort
    void sortColors(vector<int>& nums) {
        int count[3] = { 0, 0, 0 };
        for (const int i : nums) {
            count[i]++;
        }
        auto it = nums.begin();
        for (int i = 0; i < count[0]; ++i) {
            *it++ = 0;
        }
        for (int i = 0; i < count[1]; ++i) {
            *it++ = 1;
        }
        for (int i = 0; i < count[2]; ++i) {
            *it++ = 2;
        }
    }
};

File: 0075-sort-colors.py
"""
75. Sort Colors

Submitted: November 4, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 17.91 MB (beats 50.88%)
"""

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero_count = nums.count(0)
        one_count = nums.count(1)
        two_count = nums.count(2)
        nums[:] = [*[0] * zero_count, *[1] * one_count, *[2] * two_count]